### PR TITLE
ARTEMIS-1649 - enable openssl provider for Netty

### PR DESCRIPTION
**Pull request by rpahli**
We want to use the native Openssl Provider for netty to use the native openssl.
Added the supprt to switch between JDK and OpenSSL Provider.

see #1833

### CODE REVIEW

**Comment by rpahli** [comment]
I closed the PR #1833 unintentionally

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java [hunk 1]
```diff
@@ -416,6 +421,13 @@ private String getHttpUpgradeInfo() {
       return ", activemqServerName=" + serverName + ", httpUpgradeEndpoint=" + acceptor;
    }
 
+   private String realKeyStorePath;
```
Comment (michaelandrepearce) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java]: copying comment from old PR, please can we keep these method scope.

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java [hunk 2]
```diff
@@ -530,48 +539,34 @@ public synchronized void start() {
...
-      if (context != null && useServlet) {
-         // TODO: Fix me
-         //bootstrap.setOption("sslContext", context);
-      }
+      //if (context != null && useServlet) {
+      // TODO: Fix me
+      //bootstrap.setOption("sslContext", context);
+      //}
 
       bootstrap.handler(new ChannelInitializer<Channel>() {
```
Comment (michaelandrepearce) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java]: this probably could be made a lamda.

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java [hunk 3]
```diff
@@ -530,48 +539,34 @@ public synchronized void start() {
...
          }
-      } else {
-         context = null; // Unused
       }
 
-      if (context != null && useServlet) {
-         // TODO: Fix me
-         //bootstrap.setOption("sslContext", context);
-      }
+      //if (context != null && useServlet) {
```
Comment (franz1981) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java]: If isn't needed anymore please remove it

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java [hunk 4]
```diff
@@ -841,7 +864,7 @@ private HttpUpgradeHandler(ChannelPipeline pipeline, HttpClientCodec httpClientC
        * HTTP upgrade response will be decode by Netty as 2 objects:
        * - 1 HttpObject corresponding to the 101 SWITCHING PROTOCOL headers
        * - 1 EMPTY_LAST_CONTENT
-       *
+       * <p>
```
Comment (michaelandrepearce) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnector.java]: Un-needed change, please remove.

**Comment by jbertram** [comment]
Couple of things...

- All changes should be squashed into a single commit
- The commit message should follow the pattern described in the [Hacking Guide](https://github.com/apache/activemq-artemis/blob/master/docs/hacking-guide/en/maintainers.md#commit-messages).

**Comment by jbertram** [comment]
Could you update the SSL documentation in docs/user-manual/en/configuring-transports.md to include the new "sslProvider" parameter including a short explanation of why the JDK provider might be useful vs. the OpenSSL provider and vice-versa?  It's not clear from your commit what the functional benefit of this change actually is.

**Comment by rpahli** [comment]
Ok I'll add the new parameter to the documentation.
We need the openssl provider to use some special combinations of ciphersuites and elliptic curves.
    
see https://en.wikipedia.org/wiki/Comparison_of_TLS_implementations
    
Should I add these link to the documentation too?

**Comment by jbertram** [comment]
@rpahli, it would be appropriate to add the link to the documentation.

**Pull request closed without merging**